### PR TITLE
Fixes bug 1273538 - Fixes white spaces encoding in pagination URLs.

### PR DESCRIPTION
**Pull request by adngdb**

### CODE REVIEW

File: webapp-django/crashstats/base/templatetags/jinja_helpers.py [hunk 1]
```diff
@@ -7,6 +7,8 @@
 from django.contrib.staticfiles.storage import staticfiles_storage
 from django.core.urlresolvers import reverse
 
+from crashstats.base.utils import urlencode_dict
```
Comment (peterbe) [inline webapp-django/crashstats/base/templatetags/jinja_helpers.py]: 👍

File: webapp-django/crashstats/signature/views.py [hunk 2]
```diff
@@ -195,7 +193,7 @@ def signature_reports(request, params):
 
     context['current_url'] = '%s?%s' % (
         reverse('signature:signature_report'),
-        current_query.urlencode()
+        current_query.urlencode().replace('+', '%20')
```
Comment (peterbe) [inline webapp-django/crashstats/signature/views.py]: Perhaps consider doing `urlencode_dict(current_query)` and inside `urlencode_dict` you do a 
```python
if hasattr(thing, 'urlencode'):
    thing = urllib.urlencode()
else:
    thing = urllib.urlencode(thing, True)
# comment here WHY we have to do this
return thing.replace('+', '%20')
```
that way, the hacky `+` replace thing is only in one place.

**Comment by peterbe** [comment]
It's tempting to merge because it DOES solve the problem but I don't like that you've sprinkled the "hacky solution" in 3 places.

**Comment by adngdb** [comment]
> It's tempting to merge because it DOES solve the problem but I don't like that you've sprinkled the "hacky solution" in 3 places. 

+1

**Comment by peterbe** [comment]
Sorry for being a butt but could you do one last thing? 
Can you rename the function? `urlencode_dict` sounds like it only takes Python dict objects. 
How about simply `urlencode`?

**Comment by adngdb** [comment]
I thought about that, but I find ``urlencode`` even more confusing. Maybe ``urlencode_obj`` or ``urlencode_object``? What do you think?

**Comment by peterbe** [comment]
Or `encode_url`? Actually, this is up to you now. You decide. I'll approve whatever you prefer. Just wanted to "make you stop and think" :)

**Comment by peterbe** [comment]
r+

**Comment by adngdb** [comment]
I have to go, feel free to merge when it's green.

**Pull request merged**